src/modules/toast/sagas.ts

```diff
@@ -30,3 +50,61 @@ function* handleHideToast(action: HideToastAction) {
   const { id } = action.payload
   cache.remove(id)
 }
+
+function* handleHideAllToasts() {
+  const toasts: ReturnType<typeof getToasts> = yield select(getToasts)
+  for (const toast of toasts) {
+    yield put(hideToast(toast.id))
+  }
+}
+
+export function createMetaTransactionsErrorChannel() {
+  return eventChannel<ErrorCode>(emitter => {
+    function handleError(
+      data: TransactionEventData<TransactionEventType.ERROR>
+    ) {
+      if (data.error instanceof MetaTransactionError) {
+        emitter(data.error.code)
+      }
+    }
+    transactionEvents.on(TransactionEventType.ERROR, handleError)
+    return () =>
+      transactionEvents.removeListener(TransactionEventType.ERROR, handleError)
+  })
+}
+
+export function* watchMetaTransactionErrors() {
+  const channel: EventChannel<ErrorCode> = yield call(
+    createMetaTransactionsErrorChannel
+  )
+  while (true) {
+    const code: ErrorCode = yield take(channel)
+    yield call(handleMetaTransactionError, code)
+  }
+}
```

Comment:
I think that they're using the while loop in the example [here](https://redux-saga.js.org/docs/advanced/Channels) to catch for any possible error, but if you're not going to error anywhere, you can just create the channel and use `takeEvery` to listen for the channel. You could also avoid the fork if you do that.

Reply:
mm i don't think i followed how the non-forking part would work, neither why would the loop would be a problem?